submodules/branchvariable/branchvariable.js

```diff
@@ -0,0 +1,186 @@
+define(function(require) {
+	var $ = require('jquery'),
+		_ = require('lodash'),
+		monster = require('monster');
+
+	return {
+		subscribe: {
+			'callflows.fetchActions': 'branchvariableDefineAction'
+		},
+
+		branchvariableDefineAction: function(args) {
+			var self = this,
+				nodes = args.actions;
+
+			$.extend(nodes, {
+				'branch_variable[]': _.merge({
+					icon: 'arrow_sign',
+					category: self.i18n.active().oldCallflows.advanced_cat,
+					module: 'branch_variable',
+					isUsable: 'true',
+					caption: function(node) {
+						return node.getMetadata('variable', '');
+					},
+					edit: _.bind(self.branchvariableCallflowEdit, self),
+					key_caption: function(node) {
+						var key = node.key;
+
+						return _.get({
+							_: self.i18n.active().callflows.menu.default_action
+						}, key, key);
+					},
+					key_edit: _.bind(self.branchvariableCallflowKeyEdit, self)
+				}, _.pick(self.i18n.active().callflows.branchvariable, [
+					'name',
+					'tip'
+				]))
+			});
+		},
+
+		branchvariableCallflowEdit: function(node, callback) {
+			var self = this,
+				initTemplate = function(scopeSchema) {
+					return $(self.getTemplate({
+						name: 'callflowEdit',
+						data: {
+							scope: {
+								selected: node.getMetadata('scope', scopeSchema.default),
+								options: _.sortBy(scopeSchema.enum)
+							},
+							variable: _
+								.chain([
+									node.getMetadata('variable', '')
+								])
+								.flatten()
+								.join('.')
+								.value()
+						},
+						submodule: 'branchvariable'
+					}));
+				},
+				bindEvents = function($popup) {
+					var $form = $popup.find('form');
+
+					monster.ui.validate($form, {
+						rules: {
+							id: {
+								required: true
+							},
+							variable: {
+								required: true
+							}
+						}
+					});
+
+					$popup.find('#scope').on('change', function(event) {
+						event.preventDefault();
+
+						var selected = $(this).val();
+
+						$popup.find('#id').parents('.control-group').toggle(selected === 'doc');
+					});
+
+					$popup.find('#save').on('click', function(event) {
+						event.preventDefault();
+
+						if (!monster.ui.valid($form)) {
+							return;
+						}
+						var formData = monster.ui.getFormData($form.get(0));
```

Comment:
`formData.id` does not seem to be stored anywhere when the edition is completed, so when the edit popup for the branch variable module is opened again, its value is empty.

Reply:
Had to do some hacking around to allow for `id` to be conditionally added.

Since we want only one branch variable action listed, the prop will always be set and toggled between empty and a value depending on which `scope` is selected.